index.js

```diff
@@ -144,132 +127,167 @@ class Disk {
 	}
 	
 	discard(offset, length, callback) {
-		console.log('UNIMPLEMENTED: discarding', length, 'bytes at offset', offset);  // eslint-disable-line no-console
+		this._insertDiskChunk(new diskchunk.DiscardDiskChunk(offset, length));
 		callback(null);
 	}
 
 	getCapacity(callback) {
-		this._getCapacity(callback);
+		if (this.capacity !== null) {
+			callback(null, this.capacity);
+			return;
+		}
+		const self = this;
+		this._getCapacity(function(err, capacity) {
+			if (err) {
+				callback(err);
+				return;
+			}
+			self.capacity = capacity;
+			callback(null, capacity);
+		});
 	}
 
-	getStream(highWaterMark, callback) {
-		if ((typeof highWaterMark === 'function') && (callback === undefined)) {
-			callback = highWaterMark;
+	getStream() {
+		// args: ([position, [length, [highWaterMark]]], callback)
```

Comment:
I think you could use a spread for the argv here, i.e.

```js
getStream(...argv) {
  const callback = argv.pop()
  const [ position, length, highWaterMark ] = argv
  // ...
}
```

Reply:
Updated